### PR TITLE
Keep breakout and order-by columns as selectable

### LINKED ISSUE #30568: [MLv2] Make `orderableColumns` and `breakoutableColumns` mark used columns instead of excluding them
As for now, `orderableColumns` and `breakoutableColumns` return columns excluding ones already used in order-by/breakout clauses, respectively. We'd like to change that behavior, so they return all columns and mark the used ones (perhaps with a boolean flag).

That should make it easier to mirror the current column-picking experience without adding too much logic to the FE. Example:

> 1. Let’s say I have a query based on the orders table
> 2. It already has two breakouts, let’s say by “Created At” and by “Product > Category”
> 3. I click the button to add a new breakout. The column picker doesn’t show “Created At” and “Category” because they’re already used
> 4. I click the “Created At” notebook cell to edit a breakout. The column picker shows “Created At” as selected, but doesn’t show “Category” because it’s used in another breakout
> 5. I click the “Products > Category” cell. The column picker now shows “Category” as selected, but doesn’t show “Created At”

### PR DESCRIPTION
**Pull request by metamben**
Fixes #30568.

The already selected columns are kept in the lists returned by `breakoutable-columns` and `orderable-columns`, but they are marked with their position in the `breakouts` and `order-bys` list, respectively.

`display-info` called on the columns returned by `breakoutable-columns` and `orderable-columns` also returns this position information under `breakout-position` and `order-by-position`, respectively.

- - -
This change is [Reviewable](https://reviewable.io/reviews/metabase/metabase/30573)

### CODE REVIEW

**Review by kulyk — APPROVED** [review]
Works as expected 👍

File: src/metabase/lib/breakout.cljc [hunk 1]
```diff
@@ -69,12 +69,21 @@
...
-         existing-breakout? (fn [x]
-                              (some (fn [existing-breakout]
-                                      (lib.equality/= (lib.ref/ref x) existing-breakout))
-                                    existing-breakouts))
-         columns            (let [stage (lib.util/query-stage query stage-number)]
-                              (lib.metadata.calculation/visible-columns query stage-number stage))]
+   (let [indexed-breakouts (map-indexed vector (breakouts query stage-number))
+
+         breakout-pos
+         (fn [x]
```
Comment (snoe) [inline src/metabase/lib/breakout.cljc:75]: nit: this indentation is pretty hard to follow imo.

File: src/metabase/lib/util.cljc [hunk 2]
```diff
@@ -465,3 +465,20 @@
...
       new-query)))
+
+(defn with-default-effective-type
+  "Adds a default :effective-type property if it does not exist and
+  :base-type is known.
+
+  This is needed only because we have to convert queries to the Legacy
+  form.
+  The round trip conversion pMBQL -> legacy MBQL -> pMBQL loses the
+  :effective-type property, but is should be present for the frontend
```
Comment (snoe) [inline src/metabase/lib/util.cljc:476]: but is should => but it should

**Review by snoe — APPROVED** [review]

File: src/metabase/lib/util.cljc (hunk 2 above)
Comment (metamben) [reply src/metabase/lib/util.cljc]: Fixed.

File: src/metabase/lib/breakout.cljc (hunk 1 above)
Comment (metamben) [reply src/metabase/lib/breakout.cljc:75]: Changed.
